src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -170,6 +164,11 @@ private bool CheckGlobalConditions(Log log, State state)
                 return log.Fail("CopyAlwaysItemExists", "Item '{0}' has CopyToOutputDirectory set to 'Always', not up to date.", _configuredProject.UnconfiguredProject.MakeRooted(copyAlwaysItemPath));
             }
 
+
+            if (state.LastAdditionalDependentFileTimesChangedAtUtc > state.LastCheckedAtUtc)
```

Comment:
We must avoid checks against `LastCheckedAtUtc` as they are not idempotent. If the user presses <kbd>Ctrl</kbd> + <kbd>F5</kbd> then the up-to-date check is invoked twice, and the second answer will be different to the first.

This comparison should be against the last build output timestamp. Move this next to the existing check against `LastItemsChangedAtUtc`.